torch/nn/utils/parametrizations.py

```diff
@@ -20,23 +20,25 @@ def __init__(
                              'got n_power_iterations={}'.format(n_power_iterations))
         self.n_power_iterations = n_power_iterations
         self.eps = eps
-        self.register_buffer('u', None)
-        self.register_buffer('v', None)
-
         weight_mat = self._reshape_weight_to_matrix(weight)
-        self._update_vectors(weight_mat)
+        h, w = weight_mat.size()
+
+        self.register_buffer('u',
+            F.normalize(weight_mat.new_empty(h).normal_(0, 1), dim=0, eps=self.eps))
+        self.register_buffer('v',
+            F.normalize(weight_mat.new_empty(w).normal_(0, 1), dim=0, eps=self.eps))
+
+        # Start with u, v initialized to some reasonable values
+        self._update_vectors(weight_mat, 15)
```

Comment:
Maybe link to the paper where its specified, or small derivation/explanation on why its reasonable.